openerp/addons/base/ir/ir_ui_view.py

```diff
@@ -1130,7 +1130,21 @@ def _validate_custom_views(self, cr, uid, model):
 
         ids = map(itemgetter(0), cr.fetchall())
         context = dict(load_all_views=True)
-        return self._check_xml(cr, uid, ids, context=context)
+        # OpenUpgrade: set invalid custom views to inactive
+        for view_id in ids:
+            try:
+                self._check_xml(cr, uid, [view_id], context=context)
+            except AttributeError:
+                view = self.browse(cr, uid, view_id, context=context)
+                _logger.warn(
+                    "Can't render custom view %s for model %s. "
+                    "Assuming you are migrating between major versions of "
+                    "OpenERP, this view is now set to inactive. Please "
+                    "review the view contents manually after the migration.",
+                    view.xml_id, view.model)
+                self.write(
```

Comment:
Isn't bettet to do it via SQL for avoiding any ORM constraint?

Reply:
Sorry, I don't see why. And I still prefer ORM operations in Openupgrade unless for performance reasons or when such breakage actually occurs.

Reply:
For not running code like the standard one: 

https://github.com/OCA/OpenUpgrade/blob/3ea086efb36dc42dd5c15a5ae428d1a34407fa6b/openerp/addons/base/ir/ir_ui_view.py#L280-L297

or any other custom code attached to the model that can be expensive or directly blocking.

Reply:
I don't understand the aversion against using the ORM all of a sudden. Using the ORM ensures the best consistency of the data. And the code you refer to is completely harmless especially in this case. Get a grip Pedro.

Reply:
It's not aversion, but prevent possible future problems as this is affecting a general process and you can't imagine what people have made in their DBs/custom modules. That's why in this case, I prefer the SQL (although normally I would go for ORM). We haven't got also a rollback mechanism in this part if it fails, isn't it?